src/nvim/tui/tui.c

```diff
@@ -50,7 +50,7 @@ typedef struct {
   struct {
     int enable_mouse, disable_mouse;
     int enable_bracketed_paste, disable_bracketed_paste;
-    int enter_insert_mode, exit_insert_mode;
+    int enter_replace_mode, enter_insert_mode, exit_insert_mode;
```

Comment:
As far as I know, ```exit_replace_mode``` is not needed.

The the insert mode flag in ```State``` is true even in replace mode. So as soon as you exit replace mode it will trigger ```exit_insert_mode```.